exam_pp/t5_qa.py

```diff
@@ -94,21 +97,74 @@ def computeMaxBatchSize(modelConfig:PretrainedConfig)-> int:
 from abc import ABC, abstractmethod
 
         
-class HfPipeline(Enum):
-    text2text = auto()
-    textgeneration = auto()
-    llama = auto()
-    qa = auto()
+class PipelineType(Enum):
+    hf_text2text = auto()
+    hf_textgeneration = auto()
+    hf_llama = auto()
+    hf_qa = auto()
+    openai = auto()
+    vllm = auto()
 
     def __str__(self):
         return self.name
 
     @staticmethod
     def from_string(arg:str):
         try:
-            return HfPipeline[arg.upper()]
+            return PipelineType[arg.upper()]
         except KeyError:
             raise argparse.ArgumentTypeError("Invalid HfPipeline choice: %s" % arg)
+
+PIPELINE_CONSTRUCTOR_DEFAULT_ARGS = {
+    PipelineType.hf_text2text: {"use_fast": True, "batch_size": BATCH_SIZE},
+    PipelineType.hf_textgeneration: {"use_fast": True, "batch_size": BATCH_SIZE},
+    PipelineType.hf_llama: {"batch_size": BATCH_SIZE, "token": None, "device_map": "auto", "model_kwargs": {"torch_dtype": torch.bfloat16, "quantization_config": {"load_in_4bit": True}}},
+    PipelineType.hf_qa: {"use_fast": True, "batch_size": BATCH_SIZE},
+    PipelineType.openai: {},
+    PipelineType.vllm: {}
+}
+
+class RubricPipelineInitConfig(BaseModel):
+    """Arguments for constructing/configuring an LlmPipeline object"""
+    max_token_len: int = 512
+    max_output_tokens: int = 512
+    question_batch_size: int = 100
+
+
+class RubricPromptRunnerInitConfig(BaseModel):
+    """Arguments for constructing/configuring a PromptRunner object"""
+    max_token_len: int = 512
+    max_output_tokens: int = 512
+    question_batch_size: int = 100
+
+
+class PipelineConfig(BaseModel):
+    """Arguments for controlling LlmPipeline behavior"""
+
+    # arguments for AutoModel.from_pretrained(...)
+    hf_model_init : Dict[str, Any] = Field(default_factory=dict)
```

Comment:
My preference is to let device configuration live in `hf_model_init`, since the traditional way to set `device` and `device_map` is via `from_pretrained`. But I haven't made this change, since I know you didn't like this idea. But at least something has to change because there doesn't seem to be logic right now for controlling how models are to be loaded across multiple GPUs.